common/version/prover_version.go

```diff
@@ -19,3 +22,54 @@ func CheckScrollProverVersion(proverVersion string) bool {
 	// compare the `scroll_prover` version
 	return remote[2] == local[2]
 }
+
+// parseVersion takes a version string and returns its major, minor, and patch numbers.
+func parseVersion(version string) (major, minor, patch int) {
+	trimVersion := strings.TrimPrefix(version, "v")
+	if trimVersion == version {
+		log.Error("version does not start with v", "vesion", version)
+		return 0, 0, 0
+	}
+
+	versionPart := strings.SplitN(trimVersion, "-", 2)[0]
+	parts := strings.Split(versionPart, ".")
+	if len(parts) != 3 {
+		log.Error("invalid version format", "expected format", "v<major>.<minor>.<patch>", "got", version)
+		return 0, 0, 0
+	}
+
+	var err error
+	major, err = strconv.Atoi(parts[0])
+	if err != nil {
+		log.Error("invalid major version", "value", parts[0], "error", err)
+		return 0, 0, 0
+	}
+
+	minor, err = strconv.Atoi(parts[1])
+	if err != nil {
+		log.Error("invalid minor version", "value", parts[1], "error", err)
+		return 0, 0, 0
+	}
+
+	patch, err = strconv.Atoi(parts[2])
+	if err != nil {
+		log.Error("invalid patch version", "value", parts[2], "error", err)
+		return 0, 0, 0
+	}
+
+	return major, minor, patch
+}
+
+// CheckScrollRepoVersion checks if the proverVersion is at least the minimum required version.
+func CheckScrollRepoVersion(proverVersion, minVersion string) bool {
```

Comment:
How about use `https://github.com/Masterminds/semver`

Reply:
changed in [4207440](https://github.com/scroll-tech/scroll/pull/1161/commits/420744081dda148400d3d90581a497f0c41391e6)